### ISSUE TITLE
`ReadableStream` is not a subtype of `AsyncIterator`

### ISSUE THREAD

**Issue by spencerwilson** [microsoft/TypeScript#57874]
### 🔎 Search Terms

- ReadableStream

### 🕗 Version & Regression Information

I'm using 5.3.2.

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about **ReadableStream/differences in environments**

### 💻 Code

```ts
const rs: AsyncIterable<Uint8Array> = new ReadableStream<Uint8Array>();
```

### 🙁 Actual behavior

tsc reports the following error:
```
Property '[Symbol.asyncIterator]' is missing in type 'ReadableStream<Uint8Array>' but required in type 'AsyncIterable<Uint8Array>'.
```

That property should not have been missing in `ReadableStream`.

### 🙂 Expected behavior

No error, because the Streams WHATWG Living Standard documents that ReadableStream is an async iterable: https://streams.spec.whatwg.org/#rs-asynciterator

### Additional information about the issue

Admittedly, this isn't widely supported. At time of writing, it may be that just 3 runtimes support it:

<img width="777" alt="image" src="https://github.com/microsoft/TypeScript/assets/5624115/6a435acf-70ab-4420-8eb1-999aa4d6ac0b">

Live data: https://caniuse.com/mdn-api_readablestream_--asynciterator

Note: ~The browser-compat-data calls this feature "experimental", but I'm not sure what the basis for that is. See https://github.com/mdn/browser-compat-data/issues/22676~ h/t @fatcerberus for explaining this classification

**Comment by fatcerberus** [comment]
Kind of interesting that Node.js supports it since 16.5.0 (July 2021) but Chrome apparently doesn't support it at all, especially since both use V8...

Most likely it's marked experimental because it's only supported by one *browser* engine (Firefox).  Node.js and Deno are not browsers.

**Comment by Jamesernator** [comment]
> Kind of interesting that Node.js supports it since 16.5.0 (July 2021) but Chrome apparently doesn't support it at all

It will be available in the [next version of Chrome](https://chromestatus.com/feature/5143121161879552).

**Comment by MartinJohns** [comment]
Duplicate of https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1677.

**Comment by SFVII** [comment]
I locked my version of "typescript": "5.2.2" in package.json and that works and also   "@types/node": "18.17.18"

**Comment by henrymcl** [comment]
I solved it by updating to `"typescript"` to `"^5.5.3"`
